### DISCUSSION TITLE
How to get the HarmonicUnique part of the force constant file (.xml) after using the SCPH correction.

### DISCUSSION THREAD

**Discussion by ZhunyunTang** [ttadano/alamode#138] [Q&A]
Dear Dr Ttadano:
I used the SCPH correction and got the force constant file (.xml), but it does not contain the HarmonicUnique section, and I want to know if there is any way to get the HarmonicUnique section.
Sincerely,
Tang

**Comment by ttadano** [comment]
The HarmonicUnique section can be generated only via the ALM code and read by ALM when FC2XML is given in the input. This section is not necessary to run phonon calculations using ANPHON.

**Reply by ZhunyunTang99** [reply]
Thank you very much for your reply. I understand that the HarmonyUnique section is not required when using ANPHON. My goal is to extract the Hessian matrix from the scph corrected force constant file (. xml). It is precisely because of the missing HarmonicUnique that the Hessian matrix cannot be obtained. So I still want to ask, is it possible to obtain the HarmonicUnique part of the corrected force constant through ALM (or other methods)?

**Reply by ttadano** [reply]
The HarmonicUnique section only contains the symmetrically inequivalent (irreducible) sets of force constants. 
Do you need the force constants to be irreducible? 
If it is not the case, I would recommend processing `PREFIX.scph_dfc2`. 

This file contains the anharmonic correction to the second-order force constants at each temperature, for example, as follows:
```
# Temp = 0
   0   0   0    0  0   0  0      0.00193717851728688
   0   0   0    0  0   0  1    -1.75831649268676e-18
   0   0   0    0  0   0  2     1.94961888968662e-18
   0   0   0    0  0   1  0     8.63780989474151e-06
   0   0   0    0  0   1  1     -7.9064761516134e-07
   0   0   0    0  0   1  2    -7.90647615163616e-07
   0   0   0    0  1   0  0     -1.5465331276839e-18
```

The format of each line is 
`n1 n2 n3 iatom ixyz jatom jxyz delta_fc2
`

`n{1,2,3}` specifies the lattice translation vector to the cell where atom `jatom` belongs, `iatom` and `jatom` label the atoms in the primitive cell, `ixyz` and `jxyz` specify the Cartesian components (x:0, y:1, z:2) of the force constants, and delta_fc2 is the correction term in the unit of (Ry/bohr^2).

**Reply by ZhunyunTang** [reply]
Yes, my aim is to get the corrected set of constants for those symmetrically inequivalent (irreducible) forces. Is there a way to achieve this, please?

best regards
tang

**Reply by ttadano** [reply]
Unfortunately, that would not be done easily because the mapping from reducible to irreducible force constants would be necessary.
